### PR TITLE
TDC-1373 : Common lib dyn-log dataset to daikon

### PR DESCRIPTION
**Pull request by armeltd**
**What is the problem this Pull Request is trying to solve?**
 Add dynamic-log from dataset to daikon, to create a library wich use by other project.

**What is the chosen solution to this problem?**
Move the code from the dataset repo 
Use maven instead of sbt
Slight changes made to avoid importing to much dependencies from dataset repository

**Link to the JIRA issue**

 https://jira.talendforge.org/browse/TDC-1373

**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/47/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/47/)

File: daikon-scala/dynamic-log/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.scalatest</groupId>
+            <artifactId>scalatest_${scala-binary-version}</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
```
Comment (sgandon) [inline daikon-scala/dynamic-log/pom.xml]: I think this should bereaved or only set at test scope, cause this is supposed to be a library and you don't want to impose Logback as a logger. Please only use slf4j for libraries.
see https://github.com/Talend/policies/pull/34/files#diff-a3b5afe3229cbca6a71278d418cd5e7aR18

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/48/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/48/)

File: daikon-scala/dynamic-log/pom.xml (hunk 1 above)
Comment (armeltd) [reply daikon-scala/dynamic-log/pom.xml]: It is not a library to log messages but to allows admins to change programmatically the log level of the app.
The use case is to change the log level on an environement without a restart or a change in conf file.
That code is tied to Logback implem and after a quick internet search I don't think there is an easy way to make this available for all slf4j implementations as the methods used to do the it are not slf4j standards.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/50/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/50/)

File: daikon-scala/dynamic-log/pom.xml (hunk 1 above)
Comment (sgandon) [reply daikon-scala/dynamic-log/pom.xml]: fair enough.

**Pull request merged**